src/main/java/org/opentripplanner/graph_builder/linking/StreetSplitter.java

```diff
@@ -1,11 +1,834 @@
+            .filter(
+                streetEdge -> streetEdge instanceof StreetEdge &&
+                    // do not find SemiPermanentPartialStreetEdges if creating new SemiPermanentPartialStreetEdges
+                    // use only permanent StreetEdges instead
+                    (createSemiPermanentEdges
+                        ? !(streetEdge instanceof SemiPermanentPartialStreetEdge)
+                        : true)
+            )
+            .map(edge -> (StreetEdge) edge)
+            // note: not filtering by radius here as distance calculation is expensive
+            // we do that below.
+            .filter(edge -> edge.canTraverse(traverseModeSet) &&
+                // only link to edges still in the graph.
+                edge.getToVertex().getIncoming().contains(edge))
+            .collect(Collectors.toList());
+
+        // Make a map of distances to all edges.
+        final TIntDoubleMap distances = new TIntDoubleHashMap();
+        for (StreetEdge e : candidateEdges) {
+            distances.put(e.getId(), distance(vertex, e, xscale));
+        }
+
+        // Sort the list.
+        Collections.sort(candidateEdges, (o1, o2) -> {
+            double diff = distances.get(o1.getId()) - distances.get(o2.getId());
+            // A Comparator must return an integer but our distances are doubles.
+            if (diff < 0)
+                return -1;
+            if (diff > 0)
+                return 1;
+            return 0;
+        });
+
+        // find the closest candidate edges
+        if (candidateEdges.isEmpty() || distances.get(candidateEdges.get(0).getId()) > radiusDeg) {
+            // We only link to stops if we are searching for origin/destination and for that we need transitStopIndex.
+            if (destructiveSplitting || transitStopIndex == null) {
+                return false;
+            }
+            LOG.debug("No street edge was found for {}", vertex);
+            // We search for closest stops (since this is only used in origin/destination linking if no edges were found)
+            // in the same way the closest edges are found.
+            List<TransitStop> candidateStops = new ArrayList<>();
+            transitStopIndex.query(env).forEach(candidateStop ->
+                candidateStops.add((TransitStop) candidateStop)
+            );
+
+            final TIntDoubleMap stopDistances = new TIntDoubleHashMap();
+
+            for (TransitStop t : candidateStops) {
+                stopDistances.put(t.getIndex(), distance(vertex, t, xscale));
+            }
+
+            Collections.sort(candidateStops, (o1, o2) -> {
+                    double diff = stopDistances.get(o1.getIndex()) - stopDistances.get(o2.getIndex());
+                    if (diff < 0) {
+                        return -1;
+                    }
+                    if (diff > 0) {
+                        return 1;
+                    }
+                    return 0;
```

Comment:
Same comment as above sort, or the above conditionals should get braces.

Reply:
Will fix.

Reply:
Oh wait, I think you mean adding the return on the same line as the if statement. That actually looks better.